tests/semantic_analyser.cpp

```diff
@@ -985,16 +976,22 @@ kprobe:f { @x[1, "hi"] = 1; delete(@x["hi", 1]); }
                             ~~~~~~~~~~~~~~~~~~
 )");
 
+  test_error("kprobe:f { @x[0] = 1; @y[5] = 5; delete(@x, @y[5], @y[6]); }", R"(
+stdin:1:34-58: ERROR: delete() requires 2 arguments (3 provided)
```

Comment:
Also incorrect. "delete() requires 1 or 2 arguments (3 provided)"

Reply:
This error is generated by the semantic analyzer by your call spec. The `delete` call for a scalar is expanded into `delete(@x, 0)`. How should we fix this?

Reply:
The current language is "delete() takes up to 2 arguments". I see that you changed `.min_args` from 1 to 2 for "delete". Can we just leave it as 1 or does that cause other issues?

Reply:
There will never be a case where it is given 1 argument, it will always be desugared and provided a key (even if it is the default `0` key). This message is now encoded in the desugar pass.